File: drivingschoolserver/drivingschoolserver/src/main/java/com/drivingschool/service/QuizService.java
package com.drivingschool.service;

import java.util.Set;

import com.drivingschool.entity.Quiz;

public interface QuizService {

	public Quiz addQuiz(Quiz quiz);
	
	public Quiz updateQuiz(Quiz quiz);
	
	public Set<Quiz> getQuizzess();
	
	public Quiz getQuiz(Long quizId);
	
	public void deleteQuiz(Long quizId);
	
}

File: drivingfront/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private _http: HttpClient) { }

  public quizzes(){
    return this._http.get(`${baseUrl}/quiz/`);
  }

  public addQuiz(quiz){
    return this._http.post(`${baseUrl}/quiz/`, quiz);
  }

  public deleteQuiz(qId){
    return this._http.delete(`${baseUrl}/quiz/${qId}`);
  }

  public getQuiz(qId){
    return this._http.get(`${baseUrl}/quiz/${qId}`);
  }

  public updateQuiz(quiz){
    return this._http.put(`${baseUrl}/quiz/`, quiz);
  }

  public getQuizzesOfCategory(cid){
    return this._http.get(`${baseUrl}/quiz/category/${cid}`);
  }

  public getActiveQuizzes(){
    return this._http.get(`${baseUrl}/quiz/active`);
  }

  public getActiveQuizzesOfCategory(cid){
    return this._http.get(`${baseUrl}/quiz/category/active/${cid}`);
  }

}
